internal/vsp/feepayment.go

```diff
@@ -95,6 +96,7 @@ type feePayment struct {
 	timerMu sync.Mutex
 	timer   *time.Timer
 
+	log    slog.Logger
```

Comment:
the fee payment holds a pointer to the client, we can access the logger through that instead of copying the interface value for every fee payment. i don't think there is any reason that one fee payment should use a different logger from any other, and it reduces the pressure on the GC a small bit since it's less pointers that it needs to follow.

Reply:
I've made the suggested change.

I included logger in feePayment because in future I want to try and remove the circular reference between Client and feePayment, but that can wait for another day.